src/main/java/io/supertokens/webserver/api/session/VerifySessionAPI.java

```diff
@@ -98,16 +100,20 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
                 JsonObject reply = new JsonObject();
                 reply.addProperty("status", "TRY_REFRESH_TOKEN");
 
-                reply.addProperty("jwtSigningPublicKey", new Utils.PubPriKey(
-                        AccessTokenSigningKey.getInstance(main).getLatestIssuedKey().value).publicKey);
-                reply.addProperty("jwtSigningPublicKeyExpiryTime",
-                        AccessTokenSigningKey.getInstance(main).getKeyExpiryTime());
+                if (!super.getVersionFromRequest(req).equals("2.18")) {
+                    reply.addProperty("jwtSigningPublicKey", new Utils.PubPriKey(
+                            SigningKeys.getInstance(main).getLatestIssuedDynamicKey().value).publicKey);
+                    reply.addProperty("jwtSigningPublicKeyExpiryTime",
+                            SigningKeys.getInstance(main).getDynamicSigningKeyExpiryTime());
 
-                if (!super.getVersionFromRequest(req).equals("2.7")
-                        && !super.getVersionFromRequest(req).equals("2.8")) {
-                    List<KeyInfo> keys = AccessTokenSigningKey.getInstance(main).getAllKeys();
-                    JsonArray jwtSigningPublicKeyListJSON = Utils.keyListToJson(keys);
-                    reply.add("jwtSigningPublicKeyList", jwtSigningPublicKeyListJSON);
+
+                    if (!super.getVersionFromRequest(req).equals("2.7")
+                            && !super.getVersionFromRequest(req).equals("2.8")
+                            && !super.getVersionFromRequest(req).equals("2.18")) {
```

Comment:
!super.getVersionFromRequest(req).equals("2.18") is probably not right - what if we add 2.19?